src/components/core/DpEditor/DpEditor.vue

```diff
@@ -9,9 +9,11 @@
       ref="linkModal"
       @insert="insertUrl" />
     <dp-upload-modal
-      v-if="toolbar.imageButton"
+      v-if="toolbar.imageButton && tusEndpoint"
```

Comment:
While it is ok to render the modal only when all needed props are given, this also leads to a silent error if a developer is not aware of the fact she has to set tusEndpoint when using imageButton. I'd suggest to add the following in the `mounted` hook:

```
if (toolbar.imageButton ^ tusEndpoint) {
  console.warn(`DpEditor is called olny one of toolbar.imageButton or tusEndpoint set. Both must be used.`)
}
```

Note that the [^ operator (bitwise XOR) ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_XOR) is just a condensed expression of this:

```
(toolbar.imageButton && !tusEndpoint || !toolbar.imageButton && tusEndpoint)
```

Reply:
https://github.com/demos-europe/demosplan-ui/pull/567/commits/12995fdabbd6446d4924f8e9de9e4bb1c5638907